app/services/catalog/order_state_transition.rb

```diff
@@ -0,0 +1,34 @@
+module Catalog
+  class OrderStateTransition
+    attr_reader :state
+
+    def initialize(order_id)
+      @order = Order.find(order_id)
+    end
+
+    def process
+      @state = determine_order_state
+      @order.update!(:state => @state)
+
+      self
```

Comment:
Do we need to return anything here? I don't see anywhere you're using the return value so I think this can go away.

Reply:
@eclarizio I think so for all service objects we have to return itself so someone can fetch attributes from it. Thats the pattern we had setup after long discussion with Jason.

Reply:
@mkanoor I just think it's weird that unless you actually *are* fetching attributes from it, it's not really a big change. Instead of doing:
```
attribute = my_service.process.attribute_you_care_about
```
you just do:
```
my_service.process
attribute = my_service.attribute_you_care_about
```

The only places currently that are using some form of `.process.attribute` are controllers returning JSON. If we were to get rid of some of the various `attr_reader`s to attempt to make some of the attributes more private and then instead of returning `self` we were returning the json representation of the final state of processing, that would make more sense to me.

I guess my lengthy response here might suggest otherwise, but, I don't really care that much, if that's the standard we've decided on, I just found it a bit odd.

/shrug

Reply:
@eclarizio We have gone around this issue multiple times in multiple PR's. Jason didn't want the service objects to know what to return. The caller usually the controller would decide what to pull from the service object and convert it to JSON and send it out.